src/components/composer/grid_composer/grid_composer.ts

```diff
@@ -43,53 +37,58 @@ export class GridComposer extends Component<Props, SpreadsheetChildEnv> {
   static template = "o-spreadsheet-GridComposer";
   static components = { Composer };
 
-  private gridComposerRef!: Ref<HTMLElement>;
-
-  private zone!: Zone;
-  private rect!: Rect;
+  private rect: Rect = { x: 0, y: 0, width: 0, height: 0 };
+  private isEditing: boolean = false;
 
-  private composerState!: ComposerState;
+  get defaultRect() {
+    return { x: 0, y: 0, width: 0, height: 0 };
+  }
 
   setup() {
-    this.gridComposerRef = useRef("gridComposer");
-    this.composerState = useState({
-      rect: null,
-      delimitation: null,
-    });
-    const { col, row } = this.env.model.getters.getPosition();
-    this.zone = this.env.model.getters.expandZone(this.env.model.getters.getActiveSheetId(), {
-      left: col,
-      right: col,
-      top: row,
-      bottom: row,
-    });
-    this.rect = this.env.model.getters.getVisibleRect(this.zone);
-    onMounted(() => {
-      const el = this.gridComposerRef.el!;
-
-      //TODO Should be more correct to have a props that give the parent's clientHeight and clientWidth
-      const maxHeight = el.parentElement!.clientHeight - this.rect.y - SCROLLBAR_HIGHT;
-      el.style.maxHeight = (maxHeight + "px") as string;
-
-      const maxWidth = el.parentElement!.clientWidth - this.rect.x - SCROLLBAR_WIDTH;
-      el.style.maxWidth = (maxWidth + "px") as string;
-
-      this.composerState.rect = {
-        x: this.rect.x,
-        y: this.rect.y,
-        width: el!.clientWidth,
-        height: el!.clientHeight,
-      };
-      this.composerState.delimitation = {
-        width: el!.parentElement!.clientWidth,
-        height: el!.parentElement!.clientHeight,
-      };
+    onWillUpdateProps(() => {
+      const isEditing = this.env.model.getters.getEditionMode() !== "inactive";
+      if (this.isEditing !== isEditing) {
+        this.isEditing = isEditing;
+        if (!isEditing) {
+          this.rect = this.defaultRect;
+          return;
+        }
+        const position = this.env.model.getters.getPosition();
+        const zone = this.env.model.getters.expandZone(
+          this.env.model.getters.getActiveSheetId(),
+          positionToZone(position)
+        );
+        this.rect = this.env.model.getters.getVisibleRect(zone);
```

Comment:
do we even need to bother checking if the edition mode changed and change this.rect at all ? Can't we juste creat `get rect(){ ...}` that does all that ? this.rect isn't in a state nor is it complex to compute

Would avoid having 2 variables `this.isEditing` and `this.rect`

Reply:
it needs to be persistent because you want it to keep having the same position (DOM-wise) when we scroll (which in turn renders the whole app, the getter would then be updated and move :/